server/manager/src/test/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFilesTest.java

```diff
@@ -0,0 +1,235 @@
+      super(closeable, tmi);
+    }
+
+  }
+
+  private static class CaptureLoader extends LoadFiles.Loader {
+
+    private static class LoadResult {
+      private final List<TabletMetadata> tablets;
+      private final Files files;
+
+      public LoadResult(List<TabletMetadata> tablets, Files files) {
+        super();
+        this.tablets = tablets;
+        this.files = files;
+      }
+
+      public List<TabletMetadata> getTablets() {
+        return tablets;
+      }
+
+      public Files getFiles() {
+        return files;
+      }
+
+    }
+
+    private final List<LoadResult> results = new ArrayList<>();
+
+    @Override
+    void load(List<TabletMetadata> tablets, Files files) {
+      results.add(new LoadResult(tablets, files));
+    }
+
+    public List<LoadResult> getLoadResults() {
+      return results;
+    }
+
+    @Override
+    long finish() {
+      return 0;
+    }
+
+  }
+
+  @Test
+  public void testFindOverlappingFiles() {
+
+    TableId tid = TableId.of("1a");
+    List<TabletMetadata> tm = new ArrayList<>();
+    tm.add(TabletMetadata.create(tid.canonical(), null, "a"));
+    for (int i = 97; i < 122; i++) {
+      tm.add(TabletMetadata.create(tid.canonical(), "" + (char) (i), "" + (char) (i + 1)));
+    }
+    tm.add(TabletMetadata.create(tid.canonical(), "z", null));
+
+    String fmtTid = FateTxId.formatTid(1234L);
+    List<TabletMetadata> tablets = LoadFiles.findOverlappingTablets(fmtTid,
+        new KeyExtent(tid, new Text("c"), null), tm.iterator());
+    assertEquals(3, tablets.size());
+    assertEquals(tm.get(0), tablets.get(0));
+    assertEquals(tm.get(1), tablets.get(1));
+    assertEquals(tm.get(2), tablets.get(2));
+
+    tablets = LoadFiles.findOverlappingTablets(fmtTid, new KeyExtent(tid, null, new Text("x")),
+        tm.iterator());
+    assertEquals(3, tablets.size());
+    assertEquals(tm.get(tm.size() - 3), tablets.get(tablets.size() - 3));
+    assertEquals(tm.get(tm.size() - 2), tablets.get(tablets.size() - 2));
+    assertEquals(tm.get(tm.size() - 1), tablets.get(tablets.size() - 1));
+
+    tablets =
+        LoadFiles.findOverlappingTablets(fmtTid, new KeyExtent(tid, null, null), tm.iterator());
+    assertEquals(tm, tablets);
+
+  }
+
+  @Test
+  public void testLoadFiles() throws Exception {
+
+    TableId tid = TableId.of("1");
+    List<TabletMetadata> tm = new ArrayList<>();
+    tm.add(TabletMetadata.create(tid.canonical(), null, "a"));
+    for (int i = 97; i < 122; i++) {
+      tm.add(TabletMetadata.create(tid.canonical(), "" + (char) (i), "" + (char) (i + 1)));
+    }
+    tm.add(TabletMetadata.create(tid.canonical(), "z", null));
+
+    TabletsMetadata tabletMeta = new TestTabletsMetadata(null, tm);
+
+    Map<KeyExtent,String> loadRanges = new HashMap<>();
+    loadRanges.put(nke(null, "c"), "f1 f2");
+    loadRanges.put(nke("c", "g"), "f2 f3");
+    loadRanges.put(nke("g", "r"), "f2 f4");
```

Comment:
Use a specific loadRange that adds in files from disjoint areas to ensure the files get assigned to a correct tablet and not just the one in order.
 
```suggestion
    loadRanges.put(nke("g", "r"), "f2 f4");
    loadRanges.put(nke("l", "m"), "f1 f4");
```

Reply:
That specific modification doesn't work. I get:
```
java.lang.IllegalStateException: Unexpected prev end row 1;o;n 1;r;g
	at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.findOverlappingTablets(LoadFiles.java:423)
	at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.loadFiles(LoadFiles.java:378)
	at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFilesTest.testLoadFiles(LoadFilesTest.java:172)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```

I think that the loadFiles method is expecting the input to be in a specific order.

Reply:
So, instead of adding a new entry that covers `l` to `n` with another entry covering `g` to `r`, you have to add them like so:
```
    loadRanges.put(nke("g", "l"), "f2 f4");
    loadRanges.put(nke("l", "n"), "f1 f2 f4 f5");
    loadRanges.put(nke("n", "r"), "f2 f4");
```
I have found that the load ranges can't be overlapping one another and they have to be on tablet boundaries.

Reply:
Ah that explains it. 
I had tested this by merging in #5341 on-top which had those additional range changes.